cloud/app/server/utils.js

```diff
@@ -124,7 +138,64 @@ const waitForClickHouse = async () => {
     }
     await wait(2000);
   }
-  throw new Error('Timeout waiting for ClickHouse to be healthy');
+  throw new Error('Timeout waiting for ClickHouse to be ready');
+};
+
+/* sets up default row level security policies in ClickHouse */
+const setupClickousePermissions = async () => {
+  await ClickHouse.client.exec({
+    query: `CREATE ROW POLICY IF NOT EXISTS default_customers ON default.* USING concat('org_', OrgId, '_user') = currentUser() TO ALL`
+  });
+
+  await ClickHouse.client.exec({
+    query: `CREATE ROW POLICY IF NOT EXISTS default_admin ON default.* USING 1 TO ${process.env.CLICKHOUSE_USER || 'default'}`
+  });
 };
 
-module.exports = { getNextInRange, getVersionRange, ensureCapabilityDB, waitForClickHouse };
+/* creates ClickHouse user for an organization with SELECT access for all dbs and tables */
+const ensureClickHouseOrgUser = async (orgId) => {
+  const orgUser = `org_${orgId}_user`;
+  // Check if user exists
+  const userExists = await ClickHouse.client.query({
+    query: `SELECT name FROM system.users WHERE name = '${orgUser}'`,
+    format: 'JSONEachRow'
+  });
+  const accountsCollection = Mongo.db.collection('accounts');
+  const users = await userExists.json();
+  if (users.length > 0) {
+    log.debug(`ClickHouse user for organization ${orgId} already exists`);
+    const password = await accountsCollection.findOne({ _id: orgId })?.clickhouseCredentials?.password;
+    log.debug(`retrieved ClickHouse credentials for organization ${orgId} from mongo: ${orgUser} / ${password}`);
```

Comment:
doesn't seem to work, I get:

```
[2025-12-06T23:34:55.079Z utils debug] creating ClickHouse user for organization cfritz : org_cfritz_user / 4k6t1n5w3m5n3q4s02s2g2f2no3c
...
[2025-12-06T23:35:06.952Z utils debug] retrieved ClickHouse credentials for organization cfritz from mongo: org_cfritz_user / undefined
```

Reply:
oh, should I even be reviewing this? Looks like you've changed this in https://github.com/transitiverobotics/transitive/pull/137/files#diff-3357780820eda11b21a4148693d097ffd17caf8356e7b59fa7d1b00a9abc4d35R169-R171.